src/gui/sdlmain.cpp

```diff
@@ -2222,11 +2219,14 @@ int main(int argc, char* argv[]) {
 	if (SDL_Init_Wrapper() < 0)
 		E_Exit("Can't init SDL %s", SDL_GetError());
 	sdl.inited = true;
+	// Once initialized, ensure we clean up SDL for all exit conditions
+	atexit(SDL_Quit);
 
 #ifndef DISABLE_JOYSTICK
 	//Initialise Joystick separately. This way we can warn when it fails instead
 	//of exiting the application
-	if( SDL_InitSubSystem(SDL_INIT_JOYSTICK) < 0 ) LOG_MSG("Failed to init joystick support");
+	if( SDL_InitSubSystem(SDL_INIT_JOYSTICK) < 0 )
```

Comment:
The line got touched, so let's adjust formatting:

    if (SDL_InitSubSystem(SDL_INIT_JOYSTICK) < 0)